### PR TITLE
Add drafting and publishing bulk action

### PR DESCRIPTION
**Pull request by JoeyStk**
### Short description
This PR adds the option to change pages, events and pois to draft or public in a bulk action

### Proposed changes

- Add release note
- Add function to change the translation status
- Add bulk action call to events, pois and pages

### Side effects

None I'm aware of

### Resolved issues

Fixes: #2337

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Review by MizukiTemma — CHANGES_REQUESTED** [review]
Thank you for the PR 💪 This is a really convenient feature 😃 

I have three suggestions:
1. It's probably more friendly to show success/error message after bulk action so one can be sure that the bulk action has been really orerated.
2. We may better check user role in the view too. The bulk publish and draft are hidden for normal users but make it sure won't harm.
3. Currently it toggles only translations of the default language independent of in which language tab the user is right now. I think this feature is needed also for other languages, especially because sometimes publication status of maschine translated content is somehow not correct......

**Comment by MizukiTemma** [comment]
And those two points are subject to discussion:

Those two behaviours should be inherited in the bulk publishing/draft too for consisteny reason.

1. New version
When a content is published or set to draft by the buttons in the form, a new version is created even if there is no change. 
2. Set all to draft 
When a content is set to draft, all the version are set to draft too.

**Comment by JoeyStk** [comment]
> And those two points are subject to discussion:
Those two behaviours should be inherited in the bulk publishing/draft too for consisteny reason, in my opinion.
New version
When a content is published or set to draft by the buttons in the form, a new version is created even if there is no change.
Set all to draft
When a content is set to draft, all the version are set to draft too.

Are there opinions about this? :smiley:

File: integreat_cms/cms/views/utils/translation_status.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        for content in selected_content:
+            translation = content.get_translation(language_slug)
+            translation.status = desired_status
+            translation.save()
+            messages.success(
+                request,
+                _("The status of {} was successfully changed to {}").format(
+                    translation, desired_status.lower()
+                ),
+            )
```
Comment (MizukiTemma) [inline integreat_cms/cms/views/utils/translation_status.py:36]: This is my suggestion to the translation problem:

```suggestion
    status = {
        "PUBLIC": "Published", 
        "DRAFT": "Draft"
    }

    if request.user.is_superuser or request.user.is_staff:
        for content in selected_content:
            translation = content.get_translation(language_slug)
            translation.status = desired_status
            translation.save()
            messages.success(
                request,
                _("The status of \"{}\" was successfully changed to \"{}\"").format(
                    translation, _(status[desired_status])
                ),
            )
```

**Review by MizukiTemma — CHANGES_REQUESTED** [review]
> I had assumed that the status would be translated because in the constants file it's already translated. Do you have an idea on how to fix this?

The problem is that the parameter `desired_status` is a string, not a variable anymore, which means `desired_status = "PUBLIC"` (or "DRAFT") not  `desired_status = PUBLIC`.

One possible solution may be putting a small dictionary and convert `PUBLIC` to `Published`and to translate this.

File: integreat_cms/cms/views/utils/translation_status.py (hunk 1 above)
Comment (MizukiTemma) [reply integreat_cms/cms/views/utils/translation_status.py:36]: Oh, and we can maybe add an error message for else-case. I know those options are not shown for non staff users but...... 😅

**Comment by JoeyStk** [comment]
Thank you very much for the suggestion. Yes, this seems to fix the issue :)

**Comment by timoludwig** [comment]
> > And those two points are subject to discussion:
> > Those two behaviours should be inherited in the bulk publishing/draft too for consisteny reason, in my opinion.
> > New version
> > When a content is published or set to draft by the buttons in the form, a new version is created even if there is no change.
> > Set all to draft
> > When a content is set to draft, all the version are set to draft too.
> 
> Are there opinions about this? :smiley:

@JoeyStk I agree with @MizukiTemma in both questions.

File: integreat_cms/cms/views/utils/translation_status.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    status = {"PUBLIC": "Published", "DRAFT": "Draft"}
+
+    if request.user.is_superuser or request.user.is_staff:
+        for content in selected_content:
+            translation = content.get_translation(language_slug)
+            translation.status = desired_status
+            translation.pk = None
+            translation.version += 1
+            if desired_status == "DRAFT":
+                translation.all_versions.update(status=desired_status)
```
Comment (JoeyStk) [inline integreat_cms/cms/views/utils/translation_status.py:35]: @MizukiTemma Is this what you had in mind? :)

**Review by MizukiTemma — APPROVED** [review]
Thank you for updates 💪 looks good 🎉

**Review by seluianova — COMMENTED** [review]
Looks good to me!
I only found a little thing:
If the page status has not actually been changed (if it was already published or already draft), the message is anyway:
 _The status of "{page}" was successfully changed to "Published/Draft"_
Do we want to handle this case separately? To show another message maybe?

**Comment by JoeyStk** [comment]
Thank you @seluianova for your review!

> Do we want to handle this case separately? To show another message maybe?

I'm fine of doing it either way. On one hand I think you're right and it would be nice to be more precise, on the other hand since this functionality is only going to be used by the service team, I think how it is at the moment should be sufficient as well. But maybe @ulliholtgrave, @osmers or @svenseeberg are more qualified to make this decision or give some insights :)

**Comment by osmers** [comment]
I agree with @JoeyStk - also it is not 100% wrong since the page(s) are still in draft/published. So it's kinda reassuring. For me there is no need to have two separate messages :)

**Comment by JoeyStk** [comment]
>  agree with @JoeyStk - also it is not 100% wrong since the page(s) are still in draft/published. So it's kinda reassuring. For me there is no need to have two separate messages :)

Thank you. @seluianova is this fine with you and if so, could you please review again? :)

**Review by seluianova — APPROVED** [review]

**Pull request merged**